### PR TITLE
Add ParallelProjectAnalysisTests.

### PR DESCRIPTION
**Pull request by nojaf**
This is a proposal to include a unit test that asserts no projects were referenced twice.

```mermaid
graph TD;
    A-->B1;
    A-->B2;
    B1-->C;
    B2-->C;
```

Calling `checker.GetBackgroundCheckResultsForFileInProject` for a file in project A should not evaluate project C twice.

### CODE REVIEW

File: tests/service/ParallelProjectAnalysisTests.fs [hunk 1]
```diff
@@ -0,0 +1,384 @@
...
+        </> "bin"
+        </> projectName
+        </> "Debug"
+        </> targetFramework
+        </> $"{projectName}.dll"
+
+    assert File.Exists(output)
+
+    let projectReferences =
+        let mapOutput (rp: FSharpProjectOptions) = $"-r:{rp.Output}"
```
Comment (nojaf) [inline tests/service/ParallelProjectAnalysisTests.fs:67]: @baronfel, I'm not exactly sure about this part.
Should the `OtherOptions` still include `-r:` to the projects they reference as `FSharpReferencedProject.FSharpReference`?

I could not get type checking to work for `A/Library.fs` without specifying `-r:` for `B1`, `B2` and `C`.

**Comment by safesparrow** [comment]
Thanks @nojaf , I somehow missed this.

I'm wondering if we should move away from putting real test projects in the codebase and instead generate them, especially if all we care about is the project graph rather than source code details.
It would make these tests much easier to write and cheaper to maintain.

Could be useful in other tests and benchmarks as well.

File: tests/service/ParallelProjectAnalysisTests.fs [hunk 2]
```diff
@@ -0,0 +1,384 @@
...
+                "--fullpaths"
+                "--flaterrors"
+                "--highentropyva+"
+                "--targetprofile:netcore"
+                "--nocopyfsharpcore"
+                "--deterministic+"
+                "--simpleresolution"
+                "--nowarn:3384"
+                "--simpleresolution"
+            |]
```
Comment (safesparrow) [inline tests/service/ParallelProjectAnalysisTests.fs:292]: Can we use a tool (eg. Ionide.ProjInfo or Buildalyzer) to generate the options?
That's what happens in https://github.com/safesparrow/fsharp-benchmark-generator .

It's not sustainable to have an object like this for every test we want to do.

I realise that `FSharp.Compiler.Service` can't depend on `Ionide.ProjInfo`, but surely we can run the options generation in a separate process (similar to `fsharp-benchmark-generator`).

Let's make adding tests easy 👍

File: tests/service/ParallelProjectAnalysisTests.fs (hunk 1 above)
Comment (safesparrow) [reply tests/service/ParallelProjectAnalysisTests.fs:67]: I'd love to use ProjInfo to generate these instead.

**Pull request closed without merging**